### PR TITLE
chore: update PSE/QFI form

### PR DESCRIPTION
**Pull request by nhsz**
This PR updates the PSE form with the requests described in the issues below

Fixes https://github.com/ethereum/esp-website/issues/358
Fixes https://github.com/ethereum/esp-website/issues/363

**Note**: do not merge yet, needs to be tested by PSE team first

**Preview:** https://deploy-preview-370--ecosystem-support.netlify.app/pse-sponsorships/apply

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -249,6 +251,9 @@ export interface PSESponsorshipsFormData extends CaptchaForm {
   eventBudgetBreakdown: string; // SF API: Proposed_Timeline__c
   eventRequestedAmount: string; // SF API: Sponsorship_Monetary_Request__c
   additionalInfo: string; // SF API: Additional_Information__c
+  // Optional fields for Quadratic Funding Initiative option
```
Comment (pettinarip) [inline src/types.ts:254]: Looking at the preview and code, it looks like these two fields are for both options now, right?

File: src/components/forms/PSESponsorshipsForm.tsx [hunk 2]
```diff
@@ -559,679 +507,1118 @@ export const PSESponsorshipsForm: FC = () => {
...
-            <FormLabel htmlFor='eventDate' mb={1}>
-              <PageText display='inline' fontSize='input'>
-                Event date
-              </PageText>
-            </FormLabel>
+          <Controller
+            name='category'
+            control={control}
+            defaultValue={{ value: '', label: '' }}
+            render={({ field: { onChange }, fieldState: { error } }) => (
```
Comment (pettinarip) [inline src/components/forms/PSESponsorshipsForm.tsx:528]: Missing required check for this field.
```suggestion
            defaultValue={{ value: '', label: '' }}
            rules={{ required: true, validate: selected => selected.value !== '' }}
            render={({ field: { onChange }, fieldState: { error } }) => (
```

**Review by pettinarip — CHANGES_REQUESTED** [review]
@nhsz did an initial test filling out only the required fields and it worked fine. But then, on a complete test (filling out all the fields) an error was thrown. Here is the submitted payload:
```json
{
    "firstName": "complete",
    "lastName": "test",
    "email": "pettinarip@ethereum.org",
    "individualOrTeamSummary": "test",
    "city": "test",
    "website": "https://test.com",
    "twitter": "test",
    "contactTelegram": "test",
    "individualOrTeam": "Team",
    "country": "AX",
    "category": "Community event",
    "company": "test",
    "eventName": "test",
    "eventDescription": "test",
    "eventDate": "2024-04-17",
    "expectedAttendees": "123",
    "targetAudience": "test",
    "eventLink": "test.com",
    "eventRequestedAmount": "123",
    "eventBudgetBreakdown": "test",
    "whyEthereum": "",
    "eventPreviousWork": "test",
    "eventTopics": "test",
    "referrals": "test",
    "pseRationale": "test",
    "confirmedSpeakers": "test",
    "confirmedSponsors": "test",
    "additionalInfo": "test",
    "eventType": {
        "value": "Hackathon",
        "label": "Hackathon"
    },
    "eventFormat": {
        "value": "Hybrid (both)",
        "label": "Hybrid (both)"
    },
    "eventLocation": "test"
}
```

Logged error
```
Apr 10, 01:25:07 PM: fe698957 ERROR  Uncaught Exception 	{"errorType":"TypeError","errorMessage":"(C ?? \"\").trim is not a function","stack":["TypeError: (C ?? \"\").trim is not a function","    at /var/task/.next/server/pages/api/pse-sponsorships.js:1:1726","    at /var/task/node_modules/jsforce/lib/promise.js:72:9","    at process.processTicksAndRejections (node:internal/process/task_queues:77:11)"]}
```

File: src/types.ts (hunk 1 above)
Comment (nhsz) [reply src/types.ts:254]: Right

**Review by pettinarip — APPROVED** [review]
LGTM @nhsz! gj!

**Pull request merged**